apps/ios/SimpleXChat/APITypes.swift

```diff
@@ -701,24 +721,28 @@ public enum SMPTestStep: String, Decodable {
     }
 }
 
-public struct SMPTestFailure: Decodable, Error {
+public struct SMPTestFailure: Decodable, Error, Equatable {
     var testStep: SMPTestStep
     var testError: AgentErrorType
 
-    var localizedDescription: String {
-        let err = String.localizedStringWithFormat(NSLocalizedString("Test failed at step %@", comment: "server test failure"), testStep.text)
+    public static func == (l: SMPTestFailure, r: SMPTestFailure) -> Bool {
+        l.testStep == r.testStep
+    }
+
+    public var localizedDescription: String {
+        let err = String.localizedStringWithFormat(NSLocalizedString("Test failed at step %@.", comment: "server test failure"), testStep.text)
         switch testError {
         case .SMP(.AUTH):
-            return err + "," + NSLocalizedString("Server requires authentication to create queues, check password", comment: "server test error")
+            return err + NSLocalizedString(" Server requires authorization to create queues, check password", comment: "server test error")
```

Comment:
The space in the beginning is strange, and easy to miss in translations
```suggestion
            return err + " " + NSLocalizedString("Server requires authorization to create queues, check password", comment: "server test error")
```